url/dirname_test.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../assert/mod.ts";
+import * as url from "./mod.ts";
+
+const TESTSUITE = [
+  [
+    "https://deno.land/std/assert/mod.ts",
+    new URL("https://deno.land/std/assert"),
+  ],
+  [
+    new URL("https://deno.land/std/assert/mod.ts"),
+    new URL("https://deno.land/std/assert"),
+  ],
+  [
+    new URL("https://deno.land/std/assert/mod.ts?foo=bar"),
+    new URL("https://deno.land/std/assert"),
+  ],
+  [new URL("https://deno.land///"), new URL("https://deno.land")],
```

Comment:
You should probably include a test with a hash in it

Reply:
Added now.